docs/source/mods/sharpe-ratio.rst

```diff
@@ -0,0 +1,224 @@
+
+This OptiMod computes the portfolio that maximizes the Sharpe ratio for the given expected returns and variances.
+
+
+Problem Specification
+---------------------
+
+Our goal is to find an investment portfolio that maximizes
+
+.. math::
+    \begin{align*}
+        \textrm{Sharpe ratio} &= \frac{\textrm{Expected return} - \textrm{Risk-free rate}}{\textrm{Standard deviation}}.
+    \end{align*}
+
+.. dropdown:: Background: Mathematical Model
+
+    Consider :math:`n` assets. Let :math:`r_f \geq 0` be the risk-free rate. Let :math:`\mu \in \mathbb{R}^n` be the vector of expected returns and let :math:`\Sigma \in \mathbb{R}^{n \times n}` be the positive semidefinite covariance matrix. We assume there exists :math:`i \in \{1, \ldots, n\}` such that :math:`\mu_i > r_f`. If not, the portfolio that maximizes the Sharpe ratio is the one consisting entirely of the risk-free asset.
+
+    We seek a portfolio of weights :math:`x` that maximizes the Sharpe ratio :math:`(\mu^\top x - r_f) / \sqrt{x^\top \Sigma x}`:
+
+    .. math::
+        \begin{alignat*}{2}
+            \max_x\ && \frac{\mu^\top x - r_f}{\sqrt{x^\top \Sigma x}} & \\
+            \textrm{s.t.}\ && \sum_{i=1}^n x_i ={} &1 \qquad \textbf{(1)} \\
+            && x \geq {} &0.
+        \end{alignat*}
+
+    Model :math:`\textbf{(1)}` is non-convex. In general, non-convex problems are very difficult to solve. Fortunately, there exists a convex reformulation of :math:`\textbf{(1)}`. As a step towards that convex reformulation, we first reformulate :math:`\textbf{(1)}` as follows:
+
+    .. math::
+        \begin{alignat*}{2}
+            \max_y\ && \frac{1}{\sqrt{y^\top \Sigma y}} \qquad & \\
+            \textrm{s.t.}\enspace && (\mu - r_f)^\top y = {} &1 \qquad \textbf{(2)} \\
+            && y \geq {} &0.
+        \end{alignat*}
+
+    Models :math:`\textbf{(1)}` and :math:`\textbf{(2)}` are equivalent in the sense that given a solution to either problem, we can construct a solution to the other of equal or better objective value. In particular, any solution :math:`\bar{x}` of :math:`\textbf{(1)}` can be mapped to a solution :math:`\bar{y}` of :math:`\textbf{(2)}` of equivalent objective value using the transformation :math:`\bar{y}_i := \bar{x}_i / \mu^\top \bar{x}` for :math:`i = 1, \ldots, n`. Conversely, any solution :math:`\bar{y}` of :math:`\textbf{(2)}` can be mapped to a solution :math:`\bar{x}` of :math:`\textbf{(1)}` of equivalent objective value using the transformation :math:`\bar{x}_i := \bar{y}_i / \sum_{j = 1}^n \bar{y}_j` for :math:`i = 1, \ldots, n`.
```

Comment:
Even though there is no mathematical ambiguity, I find fractions with `/` sometimes hard to parse. 
For example, I would find $\frac{\bar{x}_i}{\mu^\top \bar{x}}$ easier to read than $\bar{x}_i / \mu^\top \bar{x}$. This has the disadvantage that the font gets very small, though. So both versions have their merits, and I don't have a strong opinion on this.

Reply:
I have a slight preference for using `/` instead of `\frac` for inline fractions here. When using `\frac`, the text is too small for me to comfortably read.

![inline_frac](https://github.com/Gurobi/gurobi-optimods/assets/8570492/39689bda-5082-4ba1-b9ce-569396817ec0)